tests/race_conditions/utils_for_race_tests.py

```
@@ -0,0 +1,332 @@
+    """
+    barrier = threading.Barrier(len(racers))
+    results: list[Any] = [None] * len(racers)
+    errors: list[BaseException | None] = [None] * len(racers)
+
+    def _run(index: int, racer: Callable[[], Any]) -> None:
+        try:
+            # Line all the racers up on the barrier so they enter the racy
+            # window as close to simultaneously as the OS allows.
+            barrier.wait(timeout=join_timeout)
+            results[index] = racer()
+        except BaseException as exc:  # noqa: PERF203 - deliberate blanket capture
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `BaseException`

(BLE001)